examples/Cassie/cassie_state_estimator.cc

```diff
@@ -1312,40 +1231,51 @@ EventStatus CassieRbtStateEstimator::Update(const Context<double>& context,
     // We get the angular velocity directly from the IMU without filtering
     // because the magnitude of noise is about 2e-3.
+    // Rotational position
+    q = Quaterniond(ekf.getState().getRotation()).normalized();
+    estimated_fb_state[0] = q.w();
+    estimated_fb_state.segment<3>(1) = q.vec();
+    // Translational position
     r_imu_to_pelvis_global = ekf.getState().getRotation() * (-imu_pos_);
-    estimated_fb_state.head(3) =
+    estimated_fb_state.segment<3>(4) =
         ekf.getState().getPosition() + r_imu_to_pelvis_global;
-    q = Quaterniond(ekf.getState().getRotation()).normalized();
-    estimated_fb_state[3] = q.w();
-    estimated_fb_state.segment<3>(4) = q.vec();
+    // Rotational velocity
     estimated_fb_state.segment<3>(7) = imu_measurement.head(3);
+    // Translational velocity
     omega_global = ekf.getState().getRotation() * imu_measurement.head(3);
     estimated_fb_state.tail(3) = ekf.getState().getVelocity() +
-        omega_global.cross(r_imu_to_pelvis_global);
-    state->get_mutable_discrete_state().get_mutable_vector(fb_state_idx_)
-        .get_mutable_value() << estimated_fb_state;
+                                 omega_global.cross(r_imu_to_pelvis_global);
+    state->get_mutable_discrete_state()
+            .get_mutable_vector(fb_state_idx_)
+            .get_mutable_value()
+        << estimated_fb_state;
 
     // Store imu measurement
-    state->get_mutable_discrete_state().get_mutable_vector(prev_imu_idx_)
-        .get_mutable_value() << imu_measurement;
+    state->get_mutable_discrete_state()
+            .get_mutable_vector(prev_imu_idx_)
+            .get_mutable_value()
+        << imu_measurement;
     // Store current time
-    state->get_mutable_discrete_state().get_mutable_vector(time_idx_)
-        .get_mutable_value() << current_time;
+    state->get_mutable_discrete_state()
+            .get_mutable_vector(time_idx_)
+            .get_mutable_value()
+        << current_time;
   }
   return EventStatus::Succeeded();
 }
 
-
 /// Workhorse state estimation function. Given a `cassie_out_t`, compute the
 /// estimated state as an OutputVector
 /// Since it needs to map from a struct to a vector, and no assumptions on the
 /// ordering of the vector are made, utilizes index maps to make this mapping.
-void CassieRbtStateEstimator::CopyStateOut(
-    const Context<double>& context, OutputVector<double>* output) const {
-  const auto& cassie_out = this->EvalAbstractInput(
-      context, cassie_out_input_port_)->get_value<cassie_out_t>();
+void CassieStateEstimator::CopyStateOut(const Context<double>& context,
+                                        OutputVector<double>* output) const {
+  const auto& cassie_out =
+      this->EvalAbstractInput(context, cassie_out_input_port_)
+          ->get_value<cassie_out_t>();
   // There might be a better way to initialize?
-  auto data = output->get_mutable_data();  // This doesn't affect timestamp value
+  auto data =
```

Comment:
what is the purpose of zero-ing out the output vector?

Reply:
I forgot why we added this initialization. I'm guessing we didn't have imu data back then, and having some values being nan was causing trouble. 

I am taking this off, since the current code seems working fine without it.